mne/export/_edf.py

```diff
@@ -0,0 +1,225 @@
+        if data.max() > pmax:
+            raise RuntimeError(f'The maximum μV of the data {data.max()} is '
+                               f'more than the physical max passed in {pmax}.')
+        if data.min() < pmin:
+            raise RuntimeError(f'The minimum μV of the data {data.min()} is '
+                               f'less than the physical min passed in {pmin}.')
+
+    # create instance of EDF Writer
+    hdl = EDFwriter(fname, file_type, n_channels)
+
+    # set channel data
+    for idx, ch in enumerate(ch_names):
+        if physical_range == 'auto':
+            # take the channel type minimum and maximum
+            ch_type = ch_types[idx]
+            pmin, pmax = ch_types_phys_min[ch_type], ch_types_phys_max[ch_type]
+
+        for key, val in [('PhysicalMaximum', pmax),
+                         ('PhysicalMinimum', pmin),
+                         ('DigitalMaximum', digital_max),
+                         ('DigitalMinimum', digital_min),
+                         ('PhysicalDimension', phys_dims),
+                         ('SampleFrequency', out_sfreq),
+                         ('SignalLabel', ch),
+                         ('PreFilter', filter_str_info)]:
+            _try_to_set_value(hdl, key, val, channel_index=idx)
+
+    # set patient info
+    subj_info = raw.info.get('subject_info')
+    if subj_info is not None:
+        birthday = subj_info.get('birthday')
+
+        # get the full name of subject if available
+        first_name = subj_info.get('first_name')
+        last_name = subj_info.get('last_name')
+        first_name = first_name or ''
+        last_name = last_name or ''
+        joiner = ''
+        if len(first_name) and len(last_name):
+            joiner = ' '
+        name = joiner.join([first_name, last_name])
+
+        hand = subj_info.get('hand')
+        sex = subj_info.get('sex')
+
+        if birthday is not None:
+            if hdl.setPatientBirthDate(birthday[0], birthday[1],
+                                       birthday[2]) != 0:
+                raise RuntimeError(f"Setting Patient Birth Date to {birthday} "
+                                   f"returned an error")
+        for key, val in [('PatientName', name),
+                         ('PatientGender', sex),
+                         ('AdditionalPatientInfo', f'hand={hand}')]:
+            _try_to_set_value(hdl, key, val)
+
+    # set measurement date
+    meas_date = raw.info['meas_date']
+    if meas_date:
+        subsecond = meas_date.microsecond / 100.
+        if hdl.setStartDateTime(year=meas_date.year, month=meas_date.month,
+                                day=meas_date.day, hour=meas_date.hour,
+                                minute=meas_date.minute,
+                                second=meas_date.second,
+                                subsecond=subsecond) != 0:  # noqa
+            raise RuntimeError(f"Setting Start Date Time {meas_date} "
+                               f"returned an error")
```

Comment:
See: https://github.com/mne-tools/mne-python/pull/9643#discussion_r694945767

I don't think this error would ever actually occur, unless someone hacked MNE. MNE has checks for these types of metadata already in place in Raw. If any of these errors ever did occur, it would be some weird runtime bug I presume.

Reply:
Same here.